File: Components/Dashboard/CS/MVCxDashboard/Code/Data/EnergyConsumption.cs
using System;
using System.Collections.Generic;
using System.Data;

namespace DashboardMainDemo {
    public static class EnergyConsumptionDataHelper {
        public static IList<EnergyConsumptionTotalDataRow> GenerateTotal(DataSet dataSet) {
            IList<EnergyConsumptionTotalDataRow> res = new List<EnergyConsumptionTotalDataRow>();
            DataRowCollection data = dataSet.Tables["CountriesTotal"].Rows;
            foreach(DataRow row in data) {
                res.Add(new EnergyConsumptionTotalDataRow {
                    Country = (string)row["Country"],
                    Year = (DateTime)row["Year"],
                    Consumption = (double)row["Consumption"],
                    Latitude = (double)row["Latitude"],
                    Longitude = (double)row["Longitude"],
                    Production = (double)row["Production"]
                });
            }
            return res;
        }
        public static IList<EnergyConsumptionBySectorDataRow> GenerateBySector(DataSet dataSet) {
            IList<EnergyConsumptionBySectorDataRow> res = new List<EnergyConsumptionBySectorDataRow>();
            DataRowCollection data = dataSet.Tables["CountriesBySector"].Rows;
            foreach(DataRow row in data) {
                res.Add(new EnergyConsumptionBySectorDataRow {
                    Country = (string)row["Country"],
                    Sector = (string)row["Sector"],
                    Year = (DateTime)row["Year"],
                    Consumption = row["Consumption"] != DBNull.Value ? (double)row["Consumption"] : 0,
                    Latitude = (double)row["Latitude"],
                    Longitude = (double)row["Longitude"]
                });
            }
            return res;
        }
    }

    public class EnergyConsumptionTotalDataRow {
        public string Country { get; set; }
        public DateTime Year { get; set; }
        public double Consumption { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public double Production { get; set; }
    }

    public class EnergyConsumptionBySectorDataRow {
        public string Country { get; set; }
        public string Sector { get; set; }
        public DateTime Year { get; set; }
        public double Consumption { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
    }
}

File: Components/Dashboard/VB/DashboardMainDemo/Data/EnergyConsumption.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Data

Namespace DashboardMainDemo
	Public NotInheritable Class EnergyConsumptionDataHelper
		Private Sub New()
		End Sub
		Public Shared Function GenerateTotal(ByVal dataSet As DataSet) As IList(Of EnergyConsumptionTotalDataRow)
			Dim res As IList(Of EnergyConsumptionTotalDataRow) = New List(Of EnergyConsumptionTotalDataRow)()
			Dim data As DataRowCollection = dataSet.Tables("CountriesTotal").Rows
			For Each row As DataRow In data
				res.Add(New EnergyConsumptionTotalDataRow With {.Country = CStr(row("Country")), .Year = CDate(row("Year")), .Consumption = CDbl(row("Consumption")), .Latitude = CDbl(row("Latitude")), .Longitude = CDbl(row("Longitude")), .Production = CDbl(row("Production"))})
			Next row
			Return res
		End Function
		Public Shared Function GenerateBySector(ByVal dataSet As DataSet) As IList(Of EnergyConsumptionBySectorDataRow)
			Dim res As IList(Of EnergyConsumptionBySectorDataRow) = New List(Of EnergyConsumptionBySectorDataRow)()
			Dim data As DataRowCollection = dataSet.Tables("CountriesBySector").Rows
			For Each row As DataRow In data
				res.Add(New EnergyConsumptionBySectorDataRow With {.Country = CStr(row("Country")), .Sector = CStr(row("Sector")), .Year = CDate(row("Year")), .Consumption = If(row("Consumption") IsNot DBNull.Value, CDbl(row("Consumption")), 0), .Latitude = CDbl(row("Latitude")), .Longitude = CDbl(row("Longitude"))})
			Next row
			Return res
		End Function
	End Class

	Public Class EnergyConsumptionTotalDataRow
		Public Property Country() As String
		Public Property Year() As DateTime
		Public Property Consumption() As Double
		Public Property Latitude() As Double
		Public Property Longitude() As Double
		Public Property Production() As Double
	End Class

	Public Class EnergyConsumptionBySectorDataRow
		Public Property Country() As String
		Public Property Sector() As String
		Public Property Year() As DateTime
		Public Property Consumption() As Double
		Public Property Latitude() As Double
		Public Property Longitude() As Double
	End Class
End Namespace
